packages/react/src/components/Rating/Rating.base.tsx

```diff
@@ -139,8 +140,9 @@ export const RatingBase: React.FunctionComponent<IRatingProps> = React.forwardRe
           onFocus={onSelectStar}
           onClick={onSelectStar} // For Safari & Firefox on OSX
           disabled={!!(disabled || readOnly)}
-          role="presentation"
+          role={!readOnly ? 'radio' : 'presentation'}
```

Comment:
This is my bad, I just realized that for ARIA textboxes, browser are supposed to calculate the value from the content. So when it's readonly, these buttons actually need to be `aria-hidden` instead of `role="presentation"` so the value doesn't end up being "1 of 5 stars, 2 of 5 stars, 3 of 5..." 😅

The role can then always be "radio", and `aria-hidden={readonly? 'true' : undefined}` -- where it's undefined instead of false, because `aria-hidden="false"` can have some weird side effects in practice, so it's best to avoid.